### ISSUE TITLE
關於OneCycle learning_rate scheduler 無法設置多次退火

### ISSUE THREAD

**Issue by goodmight** [PaddlePaddle/PaddleOCR#7742]
@bupt906 

请提供下述完整信息以便快速定位问题/Please provide the following information to quickly locate the problem

- 版本号/Version：Paddle：  PaddleOCR： 2.6  
- 问题相关组件/Related components： 
（在PaddleOCR項目中）
**_A:_** ppocr/optimizer/learning_rate.py: class OneCycle(object); 
**_B:_** ppocr/optimizer/lr_scheduler.py: class OneCycleDecay(LRScheduler);   
ppocr/optimizer/__init__.py: build_lr_scheduler 
（在PaddlePaddle項目中）
paddle/optimizer/lr.py: class LRScheduler(object);       

- 运行指令/Command Code：
配置OneCycle學習速率規劃器來訓練檢測模型。

問題：

在 **_A_** 中，類OneCycle的原型如下：
``` python
class OneCycle(object):
    """
    One Cycle learning rate decay
    Args:
        max_lr(float): Upper learning rate boundaries
        epochs(int): total training epochs
        step_each_epoch(int): steps each epoch
        anneal_strategy(str): {‘cos’, ‘linear’} Specifies the annealing strategy: “cos” for cosine annealing, “linear” for linear annealing. 
            Default: ‘cos’
        three_phase(bool): If True, use a third phase of the schedule to annihilate the learning rate according to ‘final_div_factor’ 
            instead of modifying the second phase (the first two phases will be symmetrical about the step indicated by ‘pct_start’).
        last_epoch (int, optional):  The index of last epoch. Can be set to restart training. Default: -1, means initial learning rate.
    """

```
其中可以用來調整退火（衰減）epochs長度的兩個變量：epochs和step_each_epoch，其中epochs指總訓練次數，而step_each_epoch大概是指每個epoch更新多少次參數。

總之，在代碼 **_B_** 中，有：       
` self.total_steps = epochs * steps_per_epoch`
以及
``` python 
        self._schedule_phases = [
            {
                'end_step': float(pct_start * self.total_steps) - 1,
                'start_lr': self.initial_lr,
                'end_lr': self.max_lr,
            },
            {
                'end_step': self.total_steps - 1,
                'start_lr': self.max_lr,
                'end_lr': self.min_lr,
            },
        ]
```
對於這兩個相，以下代碼 **_B_** 中的get_lr函數，只進行了一次anneal退火：
``` python
    def get_lr(self):
        computed_lr = 0.0
        step_num = self.last_epoch

        if step_num > self.total_steps:
            raise ValueError(
                "Tried to step {} times. The specified number of total steps is {}"
                .format(step_num + 1, self.total_steps))
        start_step = 0
        for i, phase in enumerate(self._schedule_phases):
            end_step = phase['end_step']
            if step_num <= end_step or i == len(self._schedule_phases) - 1:
                pct = (step_num - start_step) / (end_step - start_step)
                computed_lr = self.anneal_func(phase['start_lr'],
                                               phase['end_lr'], pct)
                break
            start_step = phase['end_step']

        return computed_lr
```
這應該不符合OneCycle本來的意思吧。參考這裏：https://zhuanlan.zhihu.com/p/350712244
OneCycle應該是從initial_lr快速上升到達max_lr，這一過程稱之爲warmstart，隨後緩慢下降，這一過程稱爲anneal退火。這樣重複多次。來突破優化凹點。
同時爲了突破凹點，PyTorch也設置了梯度動量的參數。
現在的問題就是，以目前的OneCycle實現，無法通過提供參數來實現多次warmstart+anneal：如圖
![image](https://user-images.githubusercontent.com/28631741/192439110-ad78a0c8-f3cb-477f-a6d0-cc5da8d35ae5.png)
豎直的上升沿就是warmstart，而cosine曲綫的下降綫就是aneal退火。可以看到是可以指定多次進行的。

是否是代碼實現沒有考慮到多次退火這個問題？

**Comment by goodmight** [comment]
看那張圖裏，T_0和T_mult分別表示第一次退火的Epochs跨度，以及後續退火的跨度等比增長倍率。
這個T_0和T_mult并沒有在目前的OneCycle中實現。
可以考慮增加實現。

**Comment by goodmight** [comment]
However，這個多次退火的功能其實可以通過，多次修改配置中“Global.epochs”和“Optimizer.lr.warmup_epoch”多次從斷點啓動訓練來等效。
但是如果在程序實現的話可以減免上述手動操作步驟。

**Comment by goodmight** [comment]
傻了，沒看到是OneCycle........抱歉！

實現沒有任何問題。是我把https://zhuanlan.zhihu.com/p/350712244的CosineAnnealingLR和OneCycleLR方法混淆了
